apps/els_lsp/src/els_code_lens_suggest_spec.erl

```diff
@@ -66,13 +70,27 @@ pois(Document) ->
 %%==============================================================================
 %% Internal functions
 %%==============================================================================
--spec get_type_spec(poi(), els_typer:info() | 'no_info') -> binary().
+-spec get_type_spec(poi(), els_typer:info()) -> binary().
 get_type_spec(POI, Info) ->
-  case Info of
-    'no_info' ->
-      <<"Cannot extract specs (check logs for details)">>;
-    _ ->
-      #{id := {Function, Arity}} = POI,
-      Spec = els_typer:get_type_spec(Function, Arity, Info),
-      re:replace(Spec, ",", ", ", [global, {return, binary}])
+    #{id := {Function, Arity}} = POI,
+    Spec = els_typer:get_type_spec(Function, Arity, Info),
+    re:replace(Spec, ",", ", ", [global, {return, binary}]).
+
+-spec truncate_spec_title(binary(), number()) -> binary().
+truncate_spec_title(Spec, MaxLength) ->
+  {Encoding, _BOMLength} = unicode:bom_to_encoding(Spec),
+  CodedSpec = unicode:characters_to_list(Spec, Encoding),
+  Length = string:length(CodedSpec),
```

Comment:
I'm not an expert on unicode so trying to follow what's happening here:
- it looks like `string:length/1` also works on binaries and returns a unicode-aware length, do we need to convert to list first?
- are you sure there is a BOM in the begining of `Spec` , otherwise I'm getting strange results:
```
1> Spec = unicode:characters_to_binary([200]).
<<"È"/utf8>>
2> {Encoding, _BOMLength} = unicode:bom_to_encoding(Spec).
{latin1,0}
3> CodedSpec = unicode:characters_to_list(Spec, Encoding).
[195,136]
4> string:length(CodedSpec).
2
```
I would expect a length of 1

Reply:
haha:) neither am I. I didn't understand how `unicode:characters_to_list/2` works.  You are absolutely right, I can  use the binary directly. Fixed.